server/game/core/card/propertyMixins/UnitProperties.ts

```diff
@@ -34,6 +36,24 @@ export function WithUnitProperties<TBaseClass extends InPlayCardConstructor>(Bas
     const StatsAndDamageClass = WithDamage(WithPrintedPower(BaseClass));
 
     return class AsUnit extends StatsAndDamageClass {
+        public static registerRulesListeners(game: Game) {
```

Comment:
Can this not be handled with code belonging to the specific events in question(technically in the System classes I suppose)? At the end of their eventHandlers or something?

Reply:
This is a good question. I think functionally it would work equally well, but I think design-wise I prefer this approach because it separates out the keyword functionality from the systems and keeps the implementation attached to the class it matters for.

Also, it's very unlikely that we would ever need to create something that emits one of the relevant events other than the current systems, but if there was for some reason some other way an OnUnitEntersPlay event could be emitted, we would need to duplicate the ambush / shielded check logic there.

I think realistically it would be fine either way but I lean the direction of keeping it this way